src/HealthcareApis/HealthcareApis/HealthcareApis/SetAzureRmHealthcareApisService.cs

```diff
@@ -0,0 +1,288 @@
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter EnableSmartProxy { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ServiceNameParameterSet, HelpMessage = "HealthcareApis FhirService List of Cors Origins.")]
+        [Parameter(Mandatory = false, ParameterSetName = ResourceIdParameterSet, HelpMessage = "HealthcareApis FhirService List of Cors Origins.")]
+        [ValidateNotNullOrEmpty]
+        [ValidatePattern("^(?:(?:(?:[hH][tT][tT][pP](?:[sS]|))\\:\\/\\/(?:[a-zA-Z0-9-]+[.]?)+(?:\\:[0-9]{1,5})?|[*]))$")]
+        public string[] CorsOrigin { get; set; }
+
+        [Parameter(Mandatory = false,ParameterSetName = ServiceNameParameterSet, HelpMessage = "HealthcareApis FhirService List of Cors Headers.")]
+        [Parameter(Mandatory = false,ParameterSetName = ResourceIdParameterSet, HelpMessage = "HealthcareApis FhirService List of Cors Headers.")]
+        [ValidateNotNullOrEmpty]
+        public string[] CorsHeader { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ServiceNameParameterSet,HelpMessage = "HealthcareApis FhirService List of Cors Methods.")]
+        [Parameter(Mandatory = false,ParameterSetName = ResourceIdParameterSet,HelpMessage = "HealthcareApis FhirService List of Cors Methods.")]
+        [ValidateNotNullOrEmpty]
+        public string[] CorsMethod { get; set; }
+
+        [Parameter(Mandatory = false,ParameterSetName = ServiceNameParameterSet,HelpMessage = "HealthcareApis FhirService Cors Max Age.")]
+        [Parameter(Mandatory = false,ParameterSetName = ResourceIdParameterSet,HelpMessage = "HealthcareApis FhirService Cors Max Age.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateRange(0, 99999)]
+        public int? CorsMaxAge { get; set; }
+
+        [Parameter(Mandatory = false,ParameterSetName = ServiceNameParameterSet, HelpMessage = "HealthcareApis FhirService AllowCorsCredentials.")]
+        [Parameter(Mandatory = false, ParameterSetName = ResourceIdParameterSet, HelpMessage = "HealthcareApis FhirService AllowCorsCredentials.")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter AllowCorsCredential { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ServiceNameParameterSet, HelpMessage = "List of Access Policy Object IDs.")]
+        [Parameter(Mandatory = false, ParameterSetName = ResourceIdParameterSet, HelpMessage = "List of Access Policy Object IDs.")]
+        [ValidateNotNullOrEmpty]
+        [ValidatePattern("^(([0-9A-Fa-f]{8}[-]?(?:[0-9A-Fa-f]{4}[-]?){3}[0-9A-Fa-f]{12}){1})+$")]
+        public string[] AccessPolicyObjectId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = ServiceNameParameterSet,
+            HelpMessage = "HealthcareApis Fhir Service Account Tags.")]
+        [Parameter(
+            Mandatory = false,
+            ParameterSetName = InputObjectParameterSet,
```

Comment:
@poadhika why is `-Tag` the only parameter that's in `InputObjectParameterSet` and all of the other properties aren't in this parameter set?